### ISSUE TITLE
Bug: `setImmediate` is not defined

### ISSUE THREAD

**Issue by stephanschubert** [csstools/css-has-pseudo#3]
[`window.setImmediate` is a non-standard feature](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate).

Maybe include a polyfill like [`setAsap`](https://github.com/taylorhakes/setAsap)?

**Comment by jonathantneal** [comment]
While that may be, the MDN docs appear to inaccurately track browser support. The same for https://caniuse.com/#feat=setimmediate

Also, _is it_ not on a standards track? https://github.com/w3c/setImmediate

Are there any browsers this was not working for you in?

**Comment by stephanschubert** [comment]
Hm, I took the cautionary note on MDN at face value because I tried a latest

* Chrome Canary `Version 72.0.3622.0 (Official Build) canary (64-bit)` and
* Firefox `63.0.3 (64-bit)`

Missing in both of them.

**Comment by jonathantneal** [comment]
That’s unfortunate, as the feature was fairly well supported. The reason `setImmediate` is even used is to work around an issue in Safari where state is not properly captured. After I remove `setImmediate`, I will confirm the bug again, and then try a new strategy — `requestAnimationFrame`.

**Comment by jonathantneal** [comment]
@stephanschubert, thanks for letting me know about this. I removed the `setImmediate` calls, verified the bug in Safari, added a new, universal call to `requestAnimationFrame` and then verified the fix in Safari.

Next, I went through the other browsers — as far back as Internet Explorer 9 — to verify that the demo was still working properly. Internet Explorer 9 requires additional polyfills for `MutationObserver` and `requestAnimationFrame`.

This change is published in 0.7.0.
